pkg/apis/conditioner.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apis
+
+import (
+	"reflect"
+	"sort"
+	"time"
+
+	"github.com/knative/pkg/apis"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/knative/serving/pkg/apis/pkg/v1alpha1"
+)
+
+type Conditional interface {
+	GetConditions() []v1alpha1.Condition
+	SetConditions([]v1alpha1.Condition)
```

Comment:
`v1alpha1.Condition` should be defined in `pkg/apis` and we should add type checks that our status types implement `apis.Conditional`.

Reply:
Oh, you are versioning the Condition?  I'm skeptical this shape will change since we've lifted it verbatim from Kubernetes API guidelines.

Reply:
No no... ok, so I need the generated deep copy and I was not sure how to target the pkg/apis directory using the script in hack. So this was a work around to show that the Condition struct was not defined inside of the serving v1alpha1.